### PR TITLE
jetbrains: update

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes
Update of JetBrains IDEs to latest release 2022.3
https://www.jetbrains.com/idea/whatsnew/

I have used [update.py](https://github.com/NixOS/nixpkgs/blob/27a3fda9e86e67142c33be91f40d91f8f8fda3b2/pkgs/applications/editors/jetbrains/update.py) script for that

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/jetbrains/versions.json [hunk 1]
```diff
@@ -3,56 +3,56 @@
...
+      "build_number": "223.8214.52"
     },
     "idea-ultimate": {
       "update-channel": "IntelliJ IDEA RELEASE",
       "url-template": "https://download.jetbrains.com/idea/ideaIU-{version}-no-jbr.tar.gz",
-      "version": "2022.2.3",
+      "version": "2022.3.1",
       "sha256": "7454d7e0b8f4e3d8d805dde645d28b842101bd77aea8b29125880c592e6b8c85",
-      "url": "https://download.jetbrains.com/idea/ideaIU-2022.2.3-no-jbr.tar.gz",
+      "url": "https://download.jetbrains.com/idea/ideaIU-2022.3.1-no-jbr.tar.gz",
```
Comment (vsuharnikov) [inline pkgs/applications/editors/jetbrains/versions.json:53]: 404, also can't find a related link in https://www.jetbrains.com/idea/download/other.html
I found https://youtrack.jetbrains.com/issue/IDEA-309170:
> That's correct; starting from 2022.3, we're no longer providing no-JBR distributions and corresponding patches. 
> I'll check what we can do to make the user experience better.

**Comment by LeSuisse** [comment]
Flagging this as security related, it contains fixes for CVE-2022-46825, CVE-2022-46826, CVE-2022-46827, CVE-2022-47895 and CVE-2022-47896.

File: pkgs/applications/editors/jetbrains/versions.json (hunk 1 above)
Comment (Vinetos) [reply pkgs/applications/editors/jetbrains/versions.json:53]: Jetbrains seems to no more support with no-JBR as they are patching OpenJDK feature on their side.  
Replacing this link seems to be the best to do.

**Comment by Vinetos** [comment]
May be duplicate of #201931

**Comment by KarolWojtasiuk** [comment]
> May be duplicate of #201931

Yeah, it's a duplicate

**Pull request closed without merging**